### ISSUE TITLE
Back propagated gradient not coming from next layer

### ISSUE THREAD

**Issue by BSharp94** [andrewliao11/DNI-tensorflow#2]
In the main training loop, you fill the backprop_placeholder with the gradient derived from the true loss(after a batch is fully propogated through). Instead, the true gradient should be derive from the synthetic loss of the next layer.

**Comment by andrewliao11** [comment]
thanks for reporting! I'll go and check it
